[gha][forge] promote formerly unstable tests to stable

rustielin opened the pull request:
### Description

Some formerly "unstable" Forge tests are now stable. Promote them to stable, so that we use their signal for release. They will also be run on the `quorum-store` branch!

Also since the number of stable Forge tests is now 17 (higher than I remember!), we may need to start thinking of separating them out into dedicated workflows based on test type. This is because the maximum number of nested workflows in Github Actions is 20: https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations. We can create workflows such as `forge-stable-chaos`, `forge-stable-changing-quorum`, `forge-stable-performance`, etc to further categorize these tests. For now, I've left some comments to divide the existing tests into rough categories.

Based off of discussion: https://aptos-org.slack.com/archives/C03PKBQM2RY/p1676594096419339?thread_ts=1676592185.638859&cid=C03PKBQM2RY

### Test Plan

Let it bake in CI

Also, observing the success rate for each of those tests over the last:
* 7d: [humio link](https://cloud.us.humio.com/github/search?live=false&query=%23type%3Dgithub-event%20event_type%3Djob%20%7C%20github.job.conclusion%20!%3D%20cancelled%20github.job.conclusion%20!%3D%20skipped%20github.job.conclusion%20!%3D%20null%20github.workflow.head_branch%3Dmain%20github.job.name%3D%2F.*forge%24%2F%0A%7C%20elapsed%20%3A%3D%20(now()%20-%20%40timestamp)%20%2F%2024%20%2F%203600%20%2F%201000%0A%7C%20days_ago%20%3A%3D%20math%3Aceil(elapsed)%0A%7C%20case%20%7B%20github.job.conclusion%20%3D%20%22success%22%20%7C%20success%20%3A%3D%201%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20success%20%3A%3D%200%3B%7D%20%0A%7C%20failure%20%3A%3D%201%20-%20success%0A%0A%7C%20case%20%7B%20days_ago%20%3D%201%20%7C%20success_1d%20%3A%3D%20success%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20success_1d%20%3A%3D%200%3B%7D%20%0A%7C%20case%20%7B%20days_ago%20%3D%201%20%7C%20count_1d%20%3A%3D%201%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20count_1d%20%3A%3D%200%3B%7D%20%0A%0A%7C%20case%20%7B%20days_ago%20%3C%3D%203%20%7C%20success_3d%20%3A%3D%20success%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20success_3d%20%3A%3D%200%3B%7D%20%0A%7C%20case%20%7B%20days_ago%20%3C%3D%203%20%7C%20count_3d%20%3A%3D%201%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20count_3d%20%3A%3D%200%3B%7D%20%0A%0A%7C%20case%20%7B%20days_ago%20%3E%203%20%7C%20success_before_3d%20%3A%3D%20success%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20success_before_3d%20%3A%3D%200%3B%7D%20%0A%7C%20case%20%7B%20days_ago%20%3E%203%20%7C%20count_before_3d%20%3A%3D%201%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20count_before_3d%20%3A%3D%200%3B%7D%20%0A%0A%7C%20groupBy(%5Bgithub.job.name%5D%2C%20function%3D%5Bcount(as%3Dcount)%2C%20sum(field%3Dfailure%2C%20as%3Dfailures)%2C%20avg(field%3Dsuccess%2C%20as%3Dsuccess_rate)%2C%20sum(success_1d%2C%20as%3Dcount_success_1d)%2C%20sum(count_1d%2C%20as%3Dcount_count_1d)%2C%20sum(success_3d%2C%20as%3Dcount_success_3d)%2C%20sum(count_3d%2C%20as%3Dcount_count_3d)%2C%20sum(success_before_3d%2C%20as%3Dcount_success_before_3d)%2C%20sum(count_before_3d%2C%20as%3Dcount_count_before_3d)%5D)%20%0A%7C%20success_rate_1d%20%3A%3D%20count_success_1d%20%2F%20count_count_1d%0A%7C%20success_rate_3d%20%3A%3D%20count_success_3d%20%2F%20count_count_3d%0A%7C%20success_rate_7d_to_4d%20%3A%3D%20count_success_before_3d%20%2F%20count_count_before_3d%0A%7C%20table(%5Bgithub.job.name%2C%20count%2C%20failures%2C%20success_rate%2C%20success_rate_7d_to_4d%2C%20success_rate_3d%2C%20success_rate_1d%5D%2C%20sortby%3D%5Bsuccess_rate_1d%2C%20success_rate_3d%2C%20success_rate%5D%2C%20reverse%3Dtrue)&start=7d&tz=America%2FLos_Angeles&widgetType=table-view_)
* 30d: [humio link](https://cloud.us.humio.com/github/search?live=false&query=%23type%3Dgithub-event%20event_type%3Djob%20%7C%20github.job.conclusion%20!%3D%20cancelled%20github.job.conclusion%20!%3D%20skipped%20github.job.conclusion%20!%3D%20null%20github.workflow.head_branch%3Dmain%20github.job.name%3D%2F.*forge%24%2F%0A%7C%20elapsed%20%3A%3D%20(now()%20-%20%40timestamp)%20%2F%2024%20%2F%203600%20%2F%201000%0A%7C%20days_ago%20%3A%3D%20math%3Aceil(elapsed)%0A%7C%20case%20%7B%20github.job.conclusion%20%3D%20%22success%22%20%7C%20success%20%3A%3D%201%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20success%20%3A%3D%200%3B%7D%20%0A%7C%20failure%20%3A%3D%201%20-%20success%0A%0A%7C%20case%20%7B%20days_ago%20%3D%201%20%7C%20success_1d%20%3A%3D%20success%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20success_1d%20%3A%3D%200%3B%7D%20%0A%7C%20case%20%7B%20days_ago%20%3D%201%20%7C%20count_1d%20%3A%3D%201%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20count_1d%20%3A%3D%200%3B%7D%20%0A%0A%7C%20case%20%7B%20days_ago%20%3C%3D%203%20%7C%20success_3d%20%3A%3D%20success%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20success_3d%20%3A%3D%200%3B%7D%20%0A%7C%20case%20%7B%20days_ago%20%3C%3D%203%20%7C%20count_3d%20%3A%3D%201%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20count_3d%20%3A%3D%200%3B%7D%20%0A%0A%7C%20case%20%7B%20days_ago%20%3E%203%20%7C%20success_before_3d%20%3A%3D%20success%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20success_before_3d%20%3A%3D%200%3B%7D%20%0A%7C%20case%20%7B%20days_ago%20%3E%203%20%7C%20count_before_3d%20%3A%3D%201%3B%0A%20%20%20%20%20%20%20%20%20*%20%7C%20count_before_3d%20%3A%3D%200%3B%7D%20%0A%0A%7C%20groupBy(%5Bgithub.job.name%5D%2C%20function%3D%5Bcount(as%3Dcount)%2C%20sum(field%3Dfailure%2C%20as%3Dfailures)%2C%20avg(field%3Dsuccess%2C%20as%3Dsuccess_rate)%2C%20sum(success_1d%2C%20as%3Dcount_success_1d)%2C%20sum(count_1d%2C%20as%3Dcount_count_1d)%2C%20sum(success_3d%2C%20as%3Dcount_success_3d)%2C%20sum(count_3d%2C%20as%3Dcount_count_3d)%2C%20sum(success_before_3d%2C%20as%3Dcount_success_before_3d)%2C%20sum(count_before_3d%2C%20as%3Dcount_count_before_3d)%5D)%20%0A%7C%20success_rate_1d%20%3A%3D%20count_success_1d%20%2F%20count_count_1d%0A%7C%20success_rate_3d%20%3A%3D%20count_success_3d%20%2F%20count_count_3d%0A%7C%20success_rate_7d_to_4d%20%3A%3D%20count_success_before_3d%20%2F%20count_count_before_3d%0A%7C%20table(%5Bgithub.job.name%2C%20count%2C%20failures%2C%20success_rate%2C%20success_rate_7d_to_4d%2C%20success_rate_3d%2C%20success_rate_1d%5D%2C%20sortby%3D%5Bsuccess_rate_1d%2C%20success_rate_3d%2C%20success_rate%5D%2C%20reverse%3Dtrue)&start=30d&tz=America%2FLos_Angeles&widgetType=table-view)

<!-- Please provide us with clear details for verifying that your changes work. -->

sherry-x reviewed (approved):
When we say it's "stable" now, what's the metrics we're looking at? can we attach the link to the test plan as well?

rustielin commented:
@sherry-x we're mostly looking at the test success over time. I've updated the test plan with those dashboard links. @igor-aptos has also fixed some of the known flakiness with some tests over the last few weeks, context: https://aptos-org.slack.com/archives/C03PKBQM2RY/p1676594096419339?thread_ts=1676592185.638859&cid=C03PKBQM2RY

igor-aptos commented on .github/workflows/forge-unstable.yaml:
```diff
@@ -145,120 +109,110 @@ jobs:
+      POST_TO_SLACK: ${{ needs.determine-test-metadata.outputs.BRANCH == 'main' }} # only post to slack on main branch
 
-  run-forge-state-sync-perf-validator-test:
+  run-forge-changing-working-quorum-test-high-load:
```
not flaky any more, fixed on 2023-02-07T09:30:14Z, only 2 failures since then, both on startup (not during test)

https://cloud.us.humio.com/github/search?live=false&query=%23type%3Dgithub-event%20event_type%3Djob%7Cgithub.job.conclusion!%3Dcancelled%20github.job.conclusion!%3Dskipped%20github.job.conclusion!%3Dnull%20github.workflow.head_branch%3Dmain%20github.job.name%3D%2F.*forge%24%2F%7Cgithub.job.html_url%3D%2Faptos-core%5C%2Factions%2F%7Celapsed%20%3A%3D%20((((now())-(%40timestamp))%2F(24))%2F(3600))%2F(1000)%7Cmath%3Aceil(elapsed%2Cas%3Ddays_ago)%7Ccase%20%7B%20github.job.conclusion%3D%22success%22%7Csuccess%20%3A%3D%201%3B*%7Csuccess%20%3A%3D%200%7D%7Cfailure%20%3A%3D%20(1)-(success)%7Ccase%20%7B%20days_ago%3D1%7Csuccess_1d%20%3A%3D%20success%3B*%7Csuccess_1d%20%3A%3D%200%7D%7Ccase%20%7B%20days_ago%3D1%7Ccount_1d%20%3A%3D%201%3B*%7Ccount_1d%20%3A%3D%200%7D%7Ccase%20%7B%20days_ago%3C%3D3%7Csuccess_3d%20%3A%3D%20success%3B*%7Csuccess_3d%20%3A%3D%200%7D%7Ccase%20%7B%20days_ago%3C%3D3%7Ccount_3d%20%3A%3D%201%3B*%7Ccount_3d%20%3A%3D%200%7D%7Ccase%20%7B%20days_ago%3E3%7Csuccess_before_3d%20%3A%3D%20success%3B*%7Csuccess_before_3d%20%3A%3D%200%7D%7Ccase%20%7B%20days_ago%3E3%7Ccount_before_3d%20%3A%3D%201%3B*%7Ccount_before_3d%20%3A%3D%200%7D%20%7C%20github.job.name%3D%22run-forge-changing-working-quorum-test-high-load%20%2F%20forge%22%20%7Ctable(%5Bgithub.job.name%2C%20github.job.completed_at%2C%20github.job.conclusion%2C%20github.job.html_url%5D)%0A&start=30d&tz=America%2FLos_Angeles


https://github.com/aptos-labs/aptos-core/actions/runs/4202202025/jobs/7290064988
https://github.com/aptos-labs/aptos-core/actions/runs/4150628958/jobs/7180416126

igor-aptos commented on .github/workflows/forge-unstable.yaml:
```diff
@@ -145,120 +109,110 @@ jobs:
+      FORGE_ENABLE_FAILPOINTS: true
 
-  run-forge-state-sync-slow-processing-catching-up-test:
+  run-forge-network-partition-test:
```
not flaky any more, fixed on 2023-02-08T09:31:41Z, failed only once since then, on container creation:

https://cloud.us.humio.com/github/search?live=false&query=%23type%3Dgithub-event%20event_type%3Djob%7Cgithub.job.conclusion!%3Dcancelled%20github.job.conclusion!%3Dskipped%20github.job.conclusion!%3Dnull%20github.workflow.head_branch%3Dmain%20github.job.name%3D%2F.*forge%24%2F%7Cgithub.job.html_url%3D%2Faptos-core%5C%2Factions%2F%7Celapsed%20%3A%3D%20((((now())-(%40timestamp))%2F(24))%2F(3600))%2F(1000)%7Cmath%3Aceil(elapsed%2Cas%3Ddays_ago)%7Ccase%20%7B%20github.job.conclusion%3D%22success%22%7Csuccess%20%3A%3D%201%3B*%7Csuccess%20%3A%3D%200%7D%7Cfailure%20%3A%3D%20(1)-(success)%7Ccase%20%7B%20days_ago%3D1%7Csuccess_1d%20%3A%3D%20success%3B*%7Csuccess_1d%20%3A%3D%200%7D%7Ccase%20%7B%20days_ago%3D1%7Ccount_1d%20%3A%3D%201%3B*%7Ccount_1d%20%3A%3D%200%7D%7Ccase%20%7B%20days_ago%3C%3D3%7Csuccess_3d%20%3A%3D%20success%3B*%7Csuccess_3d%20%3A%3D%200%7D%7Ccase%20%7B%20days_ago%3C%3D3%7Ccount_3d%20%3A%3D%201%3B*%7Ccount_3d%20%3A%3D%200%7D%7Ccase%20%7B%20days_ago%3E3%7Csuccess_before_3d%20%3A%3D%20success%3B*%7Csuccess_before_3d%20%3A%3D%200%7D%7Ccase%20%7B%20days_ago%3E3%7Ccount_before_3d%20%3A%3D%201%3B*%7Ccount_before_3d%20%3A%3D%200%7D%20%7C%20github.job.name%3D%22run-forge-network-partition-test%20%2F%20forge%22%20%7Ctable(%5Bgithub.job.name%2C%20github.job.completed_at%2C%20github.job.conclusion%2C%20github.job.html_url%5D)%0A&start=30d&tz=America%2FLos_Angeles

https://github.com/aptos-labs/aptos-core/actions/runs/4172126361/jobs/7222853493

rustielin commented:
Added to stable:
* `run-forge-changing-working-quorum-test`
* `run-forge-network-partition-test`
* `run-forge-changing-working-quorum-test-high-load`
